glib/src/subclass/object.rs

```diff
@@ -241,7 +241,6 @@ impl<T: ObjectImpl> ObjectImplExt for T {
             let instance = self.get_instance();
 
             let signal_id = signal.signal_id();
-            assert!(type_.is_a(&signal_id.0));
```

Comment:
You can move this whole function to `ObjectExt` now and make it work directly on `SignalId`

Reply:
There's already an emit function ObjectExt that works with signal name instead of a signal id. I wonder how to refactor such api in a nice way. For that I have two questions:
- Do we want a way to emit a signal by name or just always require a signal id?
- Do we want the `emit_with_details` to be moved as well ? I guess yes

Reply:
> Do we want a way to emit a signal by name or just always require a signal id? Do we want to provide two variants that work with different things: signal name/signal id

I would make the one with the signal id the main implementation, and simply have the one with the name query the signal id and call the other one. It internally does that already anyway.

> Do we want the `emit_with_details` to be moved as well ? I guess yes

Yes